30-viper-drug-enrichment-bydrug.Rmd

```diff
@@ -0,0 +1,188 @@
+    code_folding: hide
+    toc: true
+    toc_float: true
+---
+
+```{r global_options, include=FALSE}
+knitr::opts_chunk$set(warning=FALSE, message=FALSE)
+```
+
+
+Here, we are evaluating the 'druggability' of viper proteins that are well correlated with variables. 
+The plots at the end of this are labeled by LVs, but the enrichment is actually based on the list of viper proteins that have a correlation of 0.5 or greater with LV expression across all tumors.
+
+# Import packages
+
+First, import packages to process and plot the data. 
+
+```{r message=FALSE, warning=FALSE}
+library(dplyr)
+library(tidyr)
+library(purrr)
+library(synapser)
+synLogin()
+library(clusterProfiler)
+library(enrichplot)
+library(ggupset)
+
+viper_res <- synTableQuery('SELECT * FROM syn21260871 where numSamps = 77')$filepath %>% 
+readr::read_csv()
+
+drug_targets <- feather::read_feather(synGet('syn20700199')$path)
+
+drug_targets <- drug_targets %>% 
+  filter(mean_pchembl > 7) %>% 
+  mutate(gene= hugo_gene) %>% 
+  select(gene, std_name) 
+```
+
+# Significance testing
+
+First, create a list of all druggable genes. We're using the drug-target explorer data, for any gene for which there is drug-target relationship with a mean_pchembl value >7, which corresponds to 1 uM. This gives us a rough approximation of druggable targets in the human genome but could be a bit under conservative.
+
+Then, we'll select the viper proteins that are 0.65 or greater correlation with latent variable expression, by latent variable (so each group of genes is based on a single latent variable expression, where each gene in the group is a highly-correlated viper protein)- this is basd on the threshold in notebook 16. 
+
+Then, we'll perform a weighted Kolmogorov-Smirnov test using the `GSEA` function from the `clusterProfiler` package to assess whether any of the viper genes are enriched in the universe of druggable genes. Here, we're treating each LV as the ranked list of viper genes (for standard GSEA, this would be the differentially expressed genes, for example) and treating the list of all drug targets as the gene set we're looking for enrichment in. 
+
+
+```{r message=FALSE, warning=FALSE}
+
+term2gene <- drug_targets %>% 
+  mutate(term = std_name) %>% 
+  select(term, gene) %>% 
+  distinct()
+
+tidy <- viper_res %>% 
+  group_by(latent_var) %>% 
+  nest() %>% 
+  mutate(data = lapply(data, function(x){
+    geneList <- x$corVal 
+    names(geneList) <- as.character(x$gene) 
+    geneList <- sort(geneList, decreasing = TRUE)
+    geneList
+  })) ##nest gene lists, looking at top 5% of genes in each LV 
+
+res <- parallel::mclapply(tidy$data, function(x){
+  GSEA(geneList = x, TERM2GENE = term2gene)
+}, mc.cores = parallel::detectCores())
+
+names(res) <- tidy$latent_var
+
+##determine if no results for a given LV
+empty_idx <- lapply(res, function(x){
+  if(nrow(x@result)==0){
+    TRUE
+  }else{
+    FALSE
+  }
+})
+
+##remove if no results
+res_complt <- res[empty_idx==F]
+
```

Comment:
Can you post a table of the the drug x LV enrichment scores and p-values to Synapse? I'd like to pull it down in the correlation analysis.